verifiers/internal/gcb/slsaprovenance/iface/statement.go

```diff
@@ -0,0 +1,29 @@
+package iface
+
+import (
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+)
+
+// Statement represents statement for a predicate type and build type.
+type Statement interface {
```

Comment:
This looks pretty similar to the gha `Provenance` interface. Is it not possible to re-use that interface somehow?

Reply:
Hoping it is, https://github.com/slsa-framework/slsa-verifier/issues/652 is tracking it. I've been hesitant to change the code too much in one PR, but this is on the TODO list. So I plan on iterating. Do you think I should attempt it in this PR?

Reply:
As long as we have some idea and we are tracking it I'm ok with that.